### ISSUE TITLE
Chromecasting with Angular, session can be initiated but returns null

### ISSUE THREAD

**Issue by dwassing** [shaka-project/shaka-player#5469]
**Have you read the [Tutorials](http://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html)?**
Certainly.

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
4.3.6

**Please ask your question**
First and foremost, apologies for the length of this question. It is rough to give a full picture of my situation without providing a lot of information.

I have after much effort and trickery managed to get as far as playing DRM content with shaka player, after following [your recommended tutorial](https://github.com/PatrickKalkman/shaka-player-angular). I know shaka player does not have typescript definitions, but it is a player that I highly desire to use in my application. Until I reached chromecast, I could work around it.

My issue is this: I am initializing the chromecast programmatically, using the UI stored in my assets folder, which I have included in my angular.json config. Anything else has not worked for me.

![image](https://github.com/shaka-project/shaka-player/assets/5759474/fd5cff4f-ca50-4ee8-a82e-2e6eee4e2bed)

Sample code, as simple and detached from other things I could make it (apologies if some things seem illogical because of it):

HTML:
```
<div id="videoContainer" #videoContainer>
  <video #videoPlayer id="videoPlayer" width="100%" height="100%" playsinline preload="auto">
  </video>
</div>
```

TS:
```
import { Component, ElementRef, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';

declare let shaka: any;
declare const chrome: any;
declare const cast: any;
@Component({
  selector: 'app-shaka-player',
  templateUrl: './shaka-player.component.html',
  styleUrls: ['./shaka-player.component.css']
})

export class ShakaPlayerComponent implements OnInit, OnDestroy {
  constructor(
      // irrelevant modules
    ) { }
  @ViewChild('videoPlayer', { static: true }) videoElementRef: ElementRef | undefined;
  @ViewChild('videoContainer', { static: true }) videoContainerRef: ElementRef | undefined;
  @Input() options: { 
     // omitted, includes SOURCE, aspect ratio, video start time etc
  }

  private videoElement: HTMLVideoElement | undefined;
  private videoContainerElement: HTMLDivElement | undefined;
  private player: any;
  private connectedStatedChangedEvt;
  private remotePlayer;
  private playerController;
  private castProxy: any;
  private shakaEventManager: any;
  private castAvailable: boolean = false;
  // several of the above ones are not needed, likely. Also omitted variables like playState
  
  private static getCastContext() {
    return cast.framework.CastContext.getInstance();
  }

  ngOnInit(): void {
    if (environment.chromeCastEnabled) this.checkIfCastApiIsAvailable();
    shaka.polyfill.installAll();
    shaka.polyfill.PatchedMediaKeysApple.install();
    // shaka.media.ManifestParser.registerParserByExtension('m3u8', muxjs.mp2t.ManifestParser);
    if (shaka.Player.isBrowserSupported()) {
      this.videoElement = this.videoElementRef?.nativeElement;
      this.videoContainerElement = this.videoContainerRef?.nativeElement;
      this.initPlayer();
    } else {
      console.error('Browser not supported!');
    }
  }
  
  private async initPlayer() {
      this.player = new shaka.Player(this.videoElement);
      this.shakaEventManager = new shaka.util.EventManager();
  
      this.videoElement.muted = this.options.muted;
      this.shakaEventManager.listen(this.player, `buffering`, (event) => {
        if (this.playerPaused) return;
        if (event.buffering === false) {
          this.playState = 2;
        } else {
          this.playState = 0;
        }
      });
      this.shakaEventManager.listen(this.videoElement, 'play', (event) => {
        this.playState = 2;
      });
      this.shakaEventManager.listen(this.videoElement, 'pause', (event) => {
        this.playState = 0;
      });
  
      if (this.videoContainerElement) {
        const ui = new shaka.ui.Overlay(
          this.player,
          this.videoContainerElement,
          this.videoElement
        );
        const controls = ui.getControls();
        const player = controls.getPlayer(); // unused
        const video = controls.getVideo(); // unused
        const uiConfig = {
          'controlPanelElements': [
            'play_pause', 'time_and_duration', 'spacer', 'mute', 'volume', 'overflow_menu', 'fullscreen', 'cast'
          ],
          // 'overflowMenuButtons': ['captions', 'quality', 'language'],
          'castReceiverAppId': '74B87E6F' //  74B87E6F is ours, 1BA79154, 8D8C71A7 both work
        };
        ui.configure(uiConfig);
        controls.addEventListener('caststatuschanged', this.onCastStatusChanged);
        this.castProxy = ui.getControls().getCastProxy();
        console.log(this.castProxy);
      }
  
      // Listen for error events.
      this.player.addEventListener('error', this.onErrorEvent);
  
      this.player.configure(...); // drm configuration etc
      this.startPlayer();
  }

  private onCastStatusChanged = (event) => {
    const newCastStatus = event[`newStatus`];
    // Handle cast status change
    console.log('The new cast status is: ' + newCastStatus);
    console.log(event);
    if (newCastStatus){
      this.loadCastMedia();
    }
  }

  private startPlayer(): void {
    // src can also be overriden to be BigBuckBunny or similar
    this.player.load(this.options.sources[0].src)
      .then(() => {
        this.videoElement?.play();
      })
      .catch((e: any) => {
        console.error(e);
      });
  }

  private loadCastMedia(): void {
    const castSession = cast.framework.CastContext.getInstance().getCurrentSession();
    console.log('castsession status: ', castSession);
    // src can also be overriden to be BigBuckBunny or similar
    var mediaInfo = new chrome.cast.media.MediaInfo(this.options.sources[0].src, 'video/mp4'); 
    var request = new chrome.cast.media.LoadRequest(mediaInfo);
    // code below never gets executed because session is NULL
    castSession.loadMedia(request).then(
      function() { console.log('Load succeed'); },
      function(errorCode) { console.log('Error code: ' + errorCode); });
  }

  private checkIfCastApiIsAvailable(): void {
    $( document ).ready(() => {
      let tries = 0;
      if (this.isChrome){
        const loadCastInterval = setInterval(() => {
          if (chrome.cast.isAvailable || tries >= 10) {
            if (chrome.cast.isAvailable){
              clearInterval(loadCastInterval);
              console.log('Cast API Available!');
              this.castAvailable = true;
              this.connectedStatedChangedEvt = cast.framework.RemotePlayerEventType.IS_CONNECTED_CHANGED;
              this.remotePlayer = new cast.framework.RemotePlayer();
              this.playerController = new cast.framework.RemotePlayerController(this.remotePlayer);
              this.playerController.addEventListener(this.connectedStatedChangedEvt, this.chromeCastEventHandler);
              this.castProxy = new shaka.cast.CastProxy(this.videoElement, this.player, ShakaPlayerComponent.getCastContext());
              console.log(ShakaPlayerComponent.getCastContext());
            } else {
              clearInterval(loadCastInterval);
              // console.log('Could not initialize Cast API. If you use Google Chrome, try refreshing your browser.');
            }
          } else {
            // console.log(' Cast API Unavailable, trying ' + (9 - tries) + ' more times.');
          }
          tries++;
        }, 1000);
      }
    });
  }

  ngOnDestroy() { 
    // clean everything etc
  }
}
```

I omitted all our DRM code, I am specifically simply trying to cast clear content for this.
I chose to not omit our appId because as the comments state, I can cast to other appIds, but not my own. The [debug tool](https://casttool.appspot.com/cactool/?appId=74B87E6F) can cast just fine to my appId with any clear source using a normal load. Additionally, I have both a VideoJS player and android app that can also cast just fine to my appId. I even dumbed the receiver app down to minimal levels and omitted all DRM/optional config just to see if I could get a clear stream through, but to no avail. I am able to initialize a cast session, but then the session still reports NULL. This is very frustrating, to say the least :( As you can see, the cast context reports NO_DEVICES_AVAILABLE.
![image](https://github.com/shaka-project/shaka-player/assets/5759474/b8130bd2-9ce1-4cdc-864f-62a1a468bc37)

When casting from videoJS, I am very able to get an actual session. The cast context finds devices, and states NOT_CONNECTED, hinting at the fact that a connection can be made. The session ID can also be retrieved by console logging it once active.

So, in conclusion:

My app ID works with the debug tool, VideoJS, and android devices. It does NOT work with Shaka, besides intializing it. A connection is established, but then the session dies flat and no load requests can be made. Other app IDs work with Shaka however, where the load request is properly put through. For anyone wanting to try to cast via their environment to my app ID, I welcome you to do so. Debugging is enabled on it. I checked the [only similar issue](https://github.com/shaka-project/shaka-player/issues/2877) I could find, but that did not help me further either.

Any ideas or insights would be great. I can elaborate on my code if necessary. Thank you for taking your time reading this.
